File: src/com/boardgamegeek/data/CollectionView.java
package com.boardgamegeek.data;

import java.util.List;


public interface CollectionView {

	public abstract void removeFilter(CollectionFilterData filter);

	public abstract void addFilter(CollectionFilterData filter);

	public abstract void setSort(int sortType);
	
	public abstract void setFilterName(String name, boolean saveName);
	
	public abstract void setFilters(List<CollectionFilterData> filters);
}

File: app/src/main/java/com/boardgamegeek/export/model/CollectionView.kt
package com.boardgamegeek.export.model

import com.google.gson.annotations.Expose

data class CollectionView(
    @Expose val name: String,
    @Expose val sortType: Int,
    @Expose val starred: Boolean,
    @Expose val filters: List<Filter>
) : Model()